### PR TITLE
Fix of ascii decode problem in TF Const with string value

### PR DESCRIPTION
**Pull request by yekruglov**
Description: Fixing UnicodeDecodeError that occurs when processing a string constant with a byte representation.
The error is caused by an attempt to re-decode an already decoded string

JIRA: 45435

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests N/A: no new operations were enabled
* [x]  Transformation tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py N/A: no new models were enabled
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list N/A: no new operations were enabled
* [x]  Supported **public** models list N/A: no new models were enabled
* [x]  New operations specification N/A: no new operations were enabled
* [x]  Guide on how to convert the **public** model N/A: no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

File: model-optimizer/mo/front/tf/extractors/utils.py [hunk 1]
```diff
@@ -72,7 +72,10 @@ def tf_tensor_content(tf_dtype, shape, pb_tensor):
         value = np.array(np.frombuffer(pb_tensor.tensor_content, type_helper[0]))
     else:
         # load typed value
-        value = np.array(type_helper[1](pb_tensor), dtype=type_helper[0])
+        try:
+            value = np.array(type_helper[1](pb_tensor), dtype=type_helper[0])
+        except UnicodeDecodeError:
```
Comment (sadolini) [inline model-optimizer/mo/front/tf/extractors/utils.py:77]: Does this exception happen for real (unexpected) issue in some case? Possibly we need some warning here, but I am not sure about it.

**Review by sadolini — APPROVED** [review]

File: model-optimizer/mo/front/tf/extractors/utils.py (hunk 1 above)
Comment (lazarevevgeny) [reply model-optimizer/mo/front/tf/extractors/utils.py:77]: We should check for the data type here. And have a try only for strings I think.

Comment (lazarevevgeny) [reply model-optimizer/mo/front/tf/extractors/utils.py:77]: And for sure we should emit warning message for the user.

Comment (yekruglov) [reply model-optimizer/mo/front/tf/extractors/utils.py:77]: @lazarevevgeny What kind of the message should be? Something like "Possible problem decoding a byte-represented string value"? I do not quite understand how to describe this.

Comment (lazarevevgeny) [reply model-optimizer/mo/front/tf/extractors/utils.py:77]: `log.error('Failed to parse a tensor with Unicode characters. Note that Inference Engine does not support string literals, so the string constant should be eliminated from the graph.', extra={'is_warning': True})`

File: model-optimizer/mo/front/tf/extractors/utils.py [hunk 2]
```diff
@@ -72,7 +72,10 @@ def tf_tensor_content(tf_dtype, shape, pb_tensor):
         value = np.array(np.frombuffer(pb_tensor.tensor_content, type_helper[0]))
     else:
         # load typed value
-        value = np.array(type_helper[1](pb_tensor), dtype=type_helper[0])
+        try:
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/tf/extractors/utils.py:75]: Add unit test covering this case.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**